openpype/hosts/fusion/plugins/publish/validate_saver_resolution.py

```diff
@@ -0,0 +1,96 @@
+import pyblish.api
+from openpype.pipeline import (
+    PublishValidationError,
+    OptionalPyblishPluginMixin,
+)
+
+from openpype.hosts.fusion.api.action import SelectInvalidAction
+from openpype.hosts.fusion.api import get_current_comp
+
+
+class ValidateSaverResolution(
+    pyblish.api.InstancePlugin, OptionalPyblishPluginMixin
+):
+    """Validate that the saver input resolution matches the projects"""
+
+    order = pyblish.api.ValidatorOrder
+    label = "Validate Saver Resolution"
+    families = ["render"]
+    hosts = ["fusion"]
+    optional = True
+    actions = [SelectInvalidAction]
+
+    @classmethod
+    def get_invalid(cls, instance, wrong_resolution=None):
+        if wrong_resolution is None:
+            wrong_resolution = []
+
+        saver = instance[0]
+        firstFrame = saver.GetAttrs("TOOLNT_Region_Start")[1]
+        comp = get_current_comp()
+
+        # If the current saver hasn't bin rendered its input resolution
+        # hasn't bin saved. To combat this, add an expression in
+        # the comments field to read the resolution
+
+        # Create undo stack so we later can remove all changes
+        comp.StartUndo("Read resolution")
```

Comment:
Probably better to use the undo chunk context manager from lib?

Reply:
True. Updated the function so the undo chunk context manager's EndUndo accept a bool